sympy/core/numbers.py

```diff
@@ -2834,6 +2836,9 @@ def _eval_subs(self, old, new):
         if self == old:
             return new
 
+    def evalf(self, prec=None, **options):
+        return Float('inf')
```

Comment:
Is this just self? Should this not be `_eval_evalf` rather than `evalf`?

Reply:
This was so that when `_n2` calls `evalf` on `Infinity`, it would work right. This is more direct, although I suppose that would also work.

Reply:
Please understand that I'm just asking questions and no suggesting that the code needs to be done in any particular way.

As I understand it Infinity and `Float('inf')` are the same object. Without this method does something strange happen with evalf?

Reply:
Changing it to `_eval_evalf` causes import troubles. The other alternative would be to check for infinite comparisons before attempting to call `_n2` in `_cmp`, but defining `evalf` for the infinities works well enough. However, I will reimplement it as a wrapper around `_eval_evalf` which I will also redefine. This ensures consistent behavior if `_eval_evalf` is called instead

Reply:
I don't understand why the method is needed though. Is it not just returning self?

Reply:
Since a few infinite comparisons happen before sympy is fully imported, the call to `evalf` in `_n2` will cause import troubles looking for the definition elsewhere. This is more or less a workaround for that issue

Reply:
Okay that makes sense

Reply:
I have implemented the wrapper for the infinities, it is pushed